### PR TITLE
Add the /locations route, GET route handlers, tests, utility functions

### PR DESCRIPTION
**Pull request by dvwhite**
The focus of this PR was to add a `/locations` route, route handlers for GET requests sent to `/locations`, some tests and the `errDetail` utility function.

* Added the `/locations` route as an application route in **app.js**
* Added a route handler for GET requests sent to `/` and also `/:groomerId` of the `/locations` route
* Added a utility function `errDetail` that logs errors to the console as well as sending a custom error response shape with the following format:
```
  {
      message: 'There was a problem completing the required operation',
      validation: [],
      data: {},
  }
```
* Added a unit test of `errDetail` that checks that response matches the desired shape and returns a `500` code
* Added integration tests for the `/locations` route handlers for GET requests

### CODE REVIEW

File: api/utils/utils.test.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      data: {},
+    };
+    const err = new Error('There was a testing error');
+
+    errDetail(res, err);
+    expect(res.code).toBe(500);
+    expect(res.code).not.toBe(200);
+    expect(res.body.data).toEqual(expectedErr);
+    done();
+  });
```
Comment (sophiasagan) [inline api/utils/utils.test.js:36]: Great job implementing tests. Implementing tests can be tough.

File: data/seeds/002_locations.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  {
+    id: 2,
+    groomerId: 'bftqzfqkq3xzj953q64r',
+    businessName: "Gillian's Fine Pet Grooming",
+    address: faker.address.streetAddress('###'),
+    email: faker.internet.email(),
+    phoneNumber: faker.phone.phoneNumberFormat(),
+    lat: faker.finance.amount(-90, 90, 6),
+    lng: faker.finance.amount(-180, 180, 6),
+  },
```
Comment (sophiasagan) [inline data/seeds/002_locations.js:23]: Interesting way of limiting the latitude and longitude search.

**Review by sophiasagan — DISMISSED** [review]
Can we fix the linter issue to pass the tests?

File: data/seeds/002_locations.js (hunk 2 above)
Comment (dvwhite) [reply data/seeds/002_locations.js:23]: Thank you. I like your suggestion to use `faker.address.latitude()` and `faker.address.longitude()`. That will make it easier to understand what the fake data should represent.

File: api/utils/utils.test.js (hunk 1 above)
Comment (dvwhite) [reply api/utils/utils.test.js:36]: No problem! 👍

**Review by sophiasagan — APPROVED** [review]
Changes look great. Glad the tests are passing.

**Review by Cberumen51 — APPROVED** [review]
You had great patience with all the testing and all the frustrations of prettier

**Pull request merged**